WordPress/src/main/java/org/wordpress/android/JobServiceId.java

```diff
@@ -1,10 +1,63 @@
 package org.wordpress.android;
 
+import android.annotation.TargetApi;
+import android.app.job.JobInfo;
+import android.app.job.JobScheduler;
+import android.content.ComponentName;
+import android.content.Context;
+import android.os.PersistableBundle;
+
+import org.wordpress.android.util.AppLog;
+
+import java.util.Set;
+
 public class JobServiceId {
     public static final int JOB_STATS_SERVICE_ID = 8000;
     public static final int JOB_NOTIFICATIONS_UPDATE_SERVICE_ID = 7000;
     public static final int JOB_READER_SEARCH_SERVICE_ID = 5000;
     public static final int JOB_PUBLICIZE_UPDATE_SERVICE_ID = 3000;
     public static final int JOB_READER_UPDATE_SERVICE_ID = 2000;
     public static final int JOB_GCM_REG_SERVICE_ID = 1000;
+
+    @TargetApi(21)
+    public static boolean isJobServiceWithSameParamsPending(Context context, ComponentName componentName,
+                                                            PersistableBundle bundleCompare, String exceptKey) {
+        JobScheduler scheduler = (JobScheduler) context.getSystemService(Context.JOB_SCHEDULER_SERVICE);
+        boolean jobAlreadyScheduled = false;
+
+        for (JobInfo jobInfo : scheduler.getAllPendingJobs()) {
```

Comment:
Weird, the documentation says otherwise 
https://developer.android.com/reference/android/app/job/JobScheduler.html
Look for the method
```

Returns
--
List<JobInfo> | a list of all of the app's scheduled jobs. This includes jobs that are currently started as well as those that are still waiting to run.This value will never be null.

```